client/apps/game/src/ui/components/settlement/mint-village-pass-modal.tsx

```
@@ -11,73 +19,187 @@ interface MintVillagePassModalProps {
+      }
+    } catch (error) {
+      console.error("Error selecting random realm:", error);
+      setRealmCheckError("Error selecting random realm. Please try again.");
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  // Function to select the checked realm
+  const selectCheckedRealm = () => {
+    if (realmCheckResult && realmCheckResult.hasSlots) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)